src/arch/x86_64.rs

```diff
@@ -96,6 +105,17 @@ impl BootInfo for RustbootBootInfo {
 }
 
 impl RustbootBootInfo {
+    unsafe fn mk_framebuf() -> Framebuffer<'static, spin::MutexGuard<'static, LockedFramebuf>> {
+        let (cfg, buf) = unsafe {
+            // Safety: we can reasonably assume this will only be called
+            // after `arch_entry`, so if we've failed to initialize the
+            // framebuffer...things have gone horribly wrong...
+            FRAMEBUFFER.get_unchecked()
+        };
+        buf.force_unlock();
```

Comment:
It seems quite bad that we can have multiple locks on the same mutex at the same time because we're force-unlocking the mutex here. If we're going to be force-unlocking before we lock, the mutex isn't really doing anything for us.

Reply:
whoops, this was supposed to have been removed --- it was added while debugging. my bad!